catalog/cataloger_merge_test.go

```diff
@@ -1200,3 +1201,60 @@ func TestCataloger_MergeFromChildAfterMergeFromParent(t *testing.T) {
 		t.Fatalf("Merge err=%s, expected none", err)
 	}
 }
+
+func TestCataloger_Merge_Hooks(t *testing.T) {
```

Comment:
let me know what you think about the diff test code - if you think it is easier to read, update this one - max I can live with both cases or give you the code for this one.